firebase-ai/src/main/kotlin/com/google/firebase/ai/type/ImagenInlineImage.kt

```diff
@@ -36,4 +39,19 @@ internal constructor(public val data: ByteArray, public val mimeType: String) {
   public fun asBitmap(): Bitmap {
     return BitmapFactory.decodeByteArray(data, 0, data.size)
   }
+
+  @Serializable internal data class Internal(val bytesBase64Encoded: String)
+
+  internal fun toInternal(): Internal {
+    val base64 = Base64.encodeToString(data, Base64.NO_WRAP)
+    return Internal(base64)
+  }
+}
+
+@PublicPreviewAPI
+public fun Bitmap.toImagenImage(): ImagenInlineImage {
+  val byteArrayOutputStream = ByteArrayOutputStream()
+  this.compress(Bitmap.CompressFormat.PNG, 100, byteArrayOutputStream)
```

Comment:
Previously we decided to go with JPEG for these conversions, see https://github.com/firebase/firebase-android-sdk/blob/e4955d9088392da9631b968697274af38bfbe5bd/firebase-ai/src/main/kotlin/com/google/firebase/ai/type/Part.kt#L50 Any reason to go with PNG in this scenario?

Reply:
I can replace it with the 80% jpeg, or we could change it to PNG across the board.

Reply:
Let's go with JPEG for consistency. That being said, it's absolutely true that we need to provide better interfaces to allow devs to pass the raw bytes (and/or file descriptors) for better compatibility